### ISSUE TITLE
applying a branch puts project into a bad state after updating to 0.13.10

### ISSUE THREAD

**Issue by nathanwhit** [gitbutlerapp/gitbutler#5459]
### Version

0.13.10

### Operating System

macOS

### Distribution Method

dmg (Apple Silicon)

### Describe the issue

Upon updating to 0.13.10, I found that my existing workspace was unusable, with GitButler hitting errors and the project history being unable to load. I removed the project, and re-added it, but now if I apply a branch to the workspace I immediately begin hitting errors again – with the same "unrecoverable" state (the only way to get back to a working state is to remove the project and re-add it).

The first error that appears is consistently regarding one of our submodules, for instance:

```
The object c5d7930d5700dbbbedea607f9cf3e50acdc33e2d (160000) at 'std' could not be found
```
or

```
The object d12a68fc4930062c0bed26447a6b5245697e77c1 (160000) at 'suite' could not be found
```

Note that `std`and `suite` are git submodules of the repo, and the ref there is the commit of the submodule:

```
❯ git submodule status
 af4c6a1eee825f19d3b3cce74cfdd03ebe1a3b92 cli/bench/testdata/lsp_benchdata (af4c6a1e)
 d12a68fc4930062c0bed26447a6b5245697e77c1 tests/node_compat/runner/suite (v18.12.1-3-gd12a68f)
 c5d7930d5700dbbbedea607f9cf3e50acdc33e2d tests/util/std (release-2024.09.12)
 e78446e34a1921371658a5df08c71d83f50a2a2f tests/wpt/suite (epochs/daily/2024-08-28_01H)
```

Here's a video illustrating the issue, starting from adding the project fresh:

https://github.com/user-attachments/assets/4209b22e-d8b0-4a19-b349-6ef5af811368

I'm happy to provide any other info you need to help debug this

### How to reproduce

1. Clone https://github.com/nathanwhit/deno : `git clone https://github.com/nathanwhit/deno`
2. Init submodules: `git submodule update --init --recursive`
3. Open repo in GitButler
4. Apply the `node-gyp-server-repro-backup` branch

Note that it seems like the first time you apply it, it errors, but at least the project history will load and let you revert applying the branch. The next time you apply the branch, the project history will be broken as well

**Comment by Byron** [comment]
Thanks a lot for reporting, the video and the steps to reproduce the issue! 

I was able to reproduce it, albeit differently.

* First I cloned deno from its origin (as it was faster): `git clone https://github.com/denoland/deno`
* then I added the clone here as remote: `git remote add nathan https://github.com/nathanwhit/deno`
* finally I made the branch available locally: `git checkout -b node-gyp-server-repro-backup nathan/node-gyp-server-repro-backup`

Now when trying to go back to the `gitbutler/workspace` branch using the GitButler UI, it shows this error.

`The object d12a68fc4930062c0bed26447a6b5245697e77c1 (160000) at 'suite' could not be found`

----

##### Notes to self

Before that, while  meddling, a commit was marked as conflicting so I checked it out using the UI. The UI didn't realize it's edit mode for some reason and just wanted to go back to the integration branch. Doing so left the 'hidden trees' checked out locally somehow.

<img width="565" alt="Screenshot 2024-11-06 at 07 51 24" src="https://github.com/user-attachments/assets/72619661-d893-452b-8c94-9028b13154d8">

**Comment by Byron** [comment]
@Caleb-T-Owens as mentioned in my message above, I was able to easily enter that strange state again. While testing the fix in the UI I ran into it again. The problem is that I have no idea how to actually reproduce it when I actively look for it. So I will keep an eye out hoping that one day this will change. In theory, one could probably scour the entire code-base for checkout calls, and see if one of them doesn't extract the `.auto-resolution` tree.
